src/rpc/blockchain.cpp

```diff
@@ -1532,6 +1539,262 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
+            map_stats[plot_value].push_back((minfeerate == MAX_MONEY) ? 0 : minfeerate);
+        } else if (plot_value == "minfeerate_old") {
+            map_stats[plot_value].push_back((minfeerate_old == MAX_MONEY) ? 0 : minfeerate_old);
+        } else if (plot_value == "maxfeerate") {
+            map_stats[plot_value].push_back(maxfeerate);
+        } else if (plot_value == "maxfeerate_old") {
+            map_stats[plot_value].push_back(maxfeerate_old);
+        } else if (plot_value == "avgfeerate") {
+            map_stats[plot_value].push_back(CFeeRate(totalfee, total_weight).GetTruncatedFee(WITNESS_SCALE_FACTOR));
+        } else if (plot_value == "avgfeerate_old") {
+            map_stats[plot_value].push_back(CFeeRate(totalfee, total_size).GetTruncatedFee(1));
+        } else if (plot_value == "avgfee") {
+            map_stats[plot_value].push_back((block.vtx.size() > 1) ? totalfee / (block.vtx.size() - 1) : 0);
+        }
+    }
+}
+
+UniValue getblockstats(const JSONRPCRequest& request)
+{
+    std::set<std::string> valid_plot_values = {
+        "height",
+        "time",
+        "mediantime",
+        "txs",
+        "ins",
+        "outs",
+        "subsidy",
+        "totalfee",
+        "reward",
+        "utxo_increase",
+        "utxo_size_inc",
+        "total_size",
+        "total_weight",
+        "total_out",
+        "minfee",
+        "maxfee",
+        "medianfee",
+        "medianfeerate",
+        "medianfeerate_old",
+        "minfeerate",
+        "minfeerate_old",
+        "maxfeerate",
+        "maxfeerate_old",
+        "avgfeerate",
+        "avgfeerate_old",
+        "avgfee",
+    };
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 4)
+        throw std::runtime_error(
+            "getblockstats ( nStart nEnd plotValues )\n"
+            "\nCompute per block statistics for a given window.\n"
+            "\nArguments:\n"
+            "1. \"start\"      (numeric, required) The height of the block that starts the window.\n"
+            "2. \"end\"        (numeric, optional) The height of the block that ends the window (default: current tip).\n"
+            "3. \"plotvalues\"  (string,  optional) Values to plot (comma separated), default(all): " + boost::join(valid_plot_values, ",") +
+            "\nResult:\n"
+            "{\n"
+            "TODO"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getblockstats", "1000 1000 \"minfeerate,avgfeerate\"")
+            + HelpExampleRpc("getblockstats", "1000 1000 \"maxfeerate,avgfeerate\"")
+        );
+
+    LOCK(cs_main);
+
+    int start = request.params[0].get_int();
+    if (start < 1 || start > chainActive.Height()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Start block height out of range");
+    }
+
+    int end;
+    if (request.params.size() > 1) {
+        end = request.params[1].get_int();
+        if (end < 0 || end > chainActive.Height()) {
```

Comment:
Nit, negative block could mean `end = height - end` to avoid early blocks (not new concept here I believe)?